ts/src/header-validator/validate-json.ts

```diff
@@ -752,6 +755,47 @@ function eventReportWindow(
   )
 }
 
+function channelCapacity(ctx: Context, s: Source): void {
+  if (
+    ctx.vsv.maxEventLevelChannelCapacityPerSource === undefined ||
+    ctx.vsv.randomizedResponseEpsilon === undefined ||
+    ctx.vsv.triggerDataCardinality === undefined ||
+    s.maxEventLevelReports === null
+  ) {
+    // TODO: consider warning when this cannot be checked
+    return
+  }
+
+  const perTriggerDataConfigs = []
+  for (let i = 0n; i < ctx.vsv.triggerDataCardinality[ctx.sourceType]; i++) {
+    perTriggerDataConfigs.push(
```

Comment:
@csharrison PTAL at this in particular

Reply:
I read through this function and it LGTM to describe the existing spec.